app/helpers/component-argument.js

```diff
@@ -0,0 +1,20 @@
+import { helper } from '@ember/component/helper';
+
+/**
+ * Determine the type of the component argument for display
+ *
+ * @method componentArgumentDisplay
+ * @param {*} argument
+ * @return {*} The argument with the correct type for display
+ */
+export function componentArgumentDisplay([argument]) {
+  if (typeof argument === 'string') {
+    return JSON.stringify(argument).replace(/^\"|\"$/g, '');
```

Comment:
```suggestion
    // Escape any interior quotes – we will add the surrounding quotes in the template
    return argument.replace(/\"/g, '\\"');
```

Reply:
@chancancode doesn't replacing all interior quotes negate what we were trying to do with `JSON.stringify` in the first place?

Reply:
It is replacing the interior quotes with `\"` so it's escaping them. `JSON.stringify` does (basically) two things for us here: 1) escape the interior quotes, 2) add the surrounding quotes. We wanted both before. Now we only want (1) because we are just going to add the surrounding quotes in the template so that we can highlight them differently. So, now `JSON.stringify` is essentially doing too much for us and instead of running (1) + (2) and then reverting the effect of (2), we should just write the code to do (1) at this point, which is what my suggestion does.